lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -2183,7 +2183,18 @@ namespace Js
 #pragma prefast(suppress:6386)
             if (isNewCall)
             {
-                newValues[args.Info.Count] = newTarget;
+                  if (hasOverridingNewTarget || !BoundFunction::Is(targetObj))
+                  {
+                      newValues[args.Info.Count] = newTarget;
+                  }
+                  else
+                  {
+                      // in the case of the Target being a BoundFunction must forward the BoundFunction's target
+                      // as the newTarget value otherwise end up with constructed object not being an instance
+                      // of original target
+                      BoundFunction *boundFunction = (BoundFunction *) BoundFunction::UnsafeFromVar(targetObj);
```

Comment:
The cast `(BoundFunction *)` shouldn't be necessary here; `UnsafeFromVar` should already return a `BoundFunction *` instance.

Reply:
BoundFunction::UnsafeFromVar casts to JavascriptFunction, not BoundFunction I guessed that this is an error but didn’t want to change it as unsure if it could have any problematic side effects.

Reply:
Oh, good point. Looks like nobody has bothered to implement `BoundFunction::UnsafeFromVar` so it's picking up the version from `JavascriptFunction`. Now that we need it, we should add `UnsafeFromVar` to `BoundFunction` (implementation can look pretty much like `JavascriptFunction::UnsafeFromVar`: assert `Is`, then `static_cast`).